### DISCUSSION TITLE
Manually fetching data in binary data

### DISCUSSION THREAD

**Discussion by AlexMachin1997** [wojtekmaj/react-pdf#1409] [Q&A] [answered]
Hi everyone,

I want to use the react-pdf components however I want to manually fetch the data myself.

In my current project we have some functionality where we manually update the pdf resource via a file input with React Query. To make the whole experience better I want to manually control the data myself.

 Is it possible to manual provide binary data myself ?

Any comments or feedback is appreciated.

Thanks

Alex

**Comment by wojtekmaj** [comment, marked as answer]
Yes, absolutely. You can pass `file={myFile}` where `myFile` is a memoized object structured like this: `{ data: Uint8Array }`.  You can also use other kinds of binary data, but Uint8Array should work best.
